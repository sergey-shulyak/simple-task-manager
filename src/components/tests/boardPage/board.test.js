import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { columns, tickets } from '../testData';

import Board from '../../boardPage/board';

const props = {
  title: 'Sample title',
  columns,
  tickets,
  className: 'custom'
};

describe('Board', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(<Board {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
