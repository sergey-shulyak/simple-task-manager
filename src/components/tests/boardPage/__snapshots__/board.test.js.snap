// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Should render correctly 1`] = `
<div
  className="board"
>
  <h1
    className="board__title"
  >
    Sample title
  </h1>
  <div
    className="board__columns"
  >
    <Column
      boardId=""
      className="board__column"
      id="1"
      key="1"
      showModal={[Function]}
      tickets={
        Array [
          Object {
            "boardId": "42",
            "columnId": "1",
            "id": "1",
            "priority": "high",
            "title": "Test ticket 1",
            "type": "task",
          },
        ]
      }
      title="To do"
    />
    <Column
      boardId=""
      className="board__column"
      id="2"
      key="2"
      showModal={[Function]}
      tickets={
        Array [
          Object {
            "boardId": "42",
            "columnId": "2",
            "id": "2",
            "priority": "low",
            "title": "Test ticket 1",
            "type": "task",
          },
        ]
      }
      title="In progress"
    />
    <Column
      boardId=""
      className="board__column"
      id="3"
      key="3"
      showModal={[Function]}
      tickets={
        Array [
          Object {
            "boardId": "42",
            "columnId": "3",
            "id": "3",
            "priority": "medium",
            "title": "Test ticket 1",
            "type": "task",
          },
        ]
      }
      title="Done"
    />
  </div>
</div>
`;
